src/scheduler/resource_resv.cpp

```diff
@@ -590,28 +528,36 @@ dup_resource_resv_array(resource_resv **oresresv_arr,
  *
  */
 resource_resv *
-dup_resource_resv(resource_resv *oresresv, server_info *nsinfo, queue_info *nqinfo, schd_error *err)
+dup_resource_resv(resource_resv *oresresv, server_info *nsinfo, queue_info *nqinfo, const std::string& name)
 {
 	resource_resv *nresresv;
+	schd_error *err;
 
-	if (oresresv == NULL || nsinfo == NULL || err == NULL)
+	if (oresresv == NULL || nsinfo == NULL)
 		return NULL;
 
-	clear_schd_error(err);
+	err = new_schd_error();
+
+	if (err == NULL)
+		return NULL;
 
 	if (!is_resource_resv_valid(oresresv, err)) {
 		schdlogerr(PBSEVENT_DEBUG2, PBS_EVENTCLASS_SCHED, LOG_DEBUG, oresresv->name, "Can't dup resresv", err);
 		return NULL;
 	}
 
-	nresresv = new_resource_resv();
+	if (name.empty())
+		nresresv = new resource_resv(oresresv->name.c_str());
+	else
+		nresresv = new resource_resv(name.c_str());
```

Comment:
I'm not sure I like this, why not overload dup_resource_resv without a name argument and pass it to dup_resource_resv() with oresresv->name?